onset_maproom/tests/test_calc.py

```diff
@@ -5,6 +5,65 @@
 import data_test_calc
 
 
+def test_longest_run_length():
+
+    precip = precip_sample()
```

Comment:
What do you gain by using `precip_sample` in these tests? `longest_run_length` works for any sequence of 0s and 1s, and the tests would be a lot easier to read if they looked like
```
assert calc.longest_run_length([0, 1, 1]) == 2
```
It's not quite that simple because the argument has to be a DataArray, but you could e.g.
```
def lrl(sequence):
    return calc.longest_run_length(xr.DataArray(sequence, dims='T'), 'T')

def test_longest_run_length():
    assert lrl([0, 1, 1]) == 2
```
The same helper function would work for all the tests except the 2d one.

Reply:
(Incidentally, if the DataArray has only one dimension, you might just look for sequences along that dimension instead of requiring that the caller name it. That seems to be how most numpy and xarray functions with an axis argument do it.)

Reply:
I think originally mostly because laze to create DataArrays.

If I trust xarray documentation, the behavior you describe on dim=None, axis=None doesn't apply to all functions. e.g. it applies to all axes if not specified for sum. It doesn't say what it would do for cumsum. It requires dim for bfill and diff. These last 2 make sense since you can't bfill/diff on all axes at once (maybe you could with diff which is a linear application).

So for now I've just renamed coord dim and stuck as much to what xarray's doc claims it does.

Reply:
Good research on xarray conventions/defaults. I still think in this case it would make sense to search along the lone dimension if there's only one and no dim name is specified, but it's not a blocker.